lib/active_model/serializer/array_serializer.rb

```diff
@@ -13,7 +14,12 @@ def initialize(objects, options = {})
             :serializer,
             ActiveModel::Serializer.serializer_for(object)
           )
-          serializer_class.new(object, options.except(:serializer))
+
+          if serializer_class.nil?
+            fail NoSerializerError, "No serializer found for object: #{object.inspect}"
+          else
+            serializer_class.new(object, options.except(:serializer))
+          end
```

Comment:
We would need to update this PR if we change this behaviour at ```serializer_for```.

I think it's an edge-case. I haven't heard about ppl using AMS to renders arrays that mix serializable and non-serializable objects yet, so let's not worry about it for now :smile: